cmd/copy.go

```diff
@@ -209,19 +209,33 @@ func (raw rawCopyCmdArgs) cook() (cookedCopyCmdArgs, error) {
 	if err != nil {
 		return cooked, err
 	}
-	// User can provide either listOfFilesToCopy or include since listOFFiles mentions
-	// file names to include explicitly and include file may mention the pattern.
-	// This could conflict enumerating the files to queue up for transfer.
-	if len(raw.listOfFilesToCopy) > 0 && len(raw.legacyInclude) > 0 {
-		return cooked, fmt.Errorf("user provided argument with both listOfFilesToCopy and include flag. Only one should be provided")
-	}
 
-	// If the user provided the list of files explicitly to be copied, then parse the argument
-	// The user passes the location of json file which will have the list of files to be copied.
-	// The "json file" is chosen as input because there is limit on the number of characters that
-	// can be supplied with the argument, but Storage Explorer folks requirements was not to impose
-	// any limit on the number of files that can be copied.
-	if len(raw.listOfFilesToCopy) > 0 {
+	// Note: new implementation of list-of-files only works for remove command for now
+	if raw.fromTo == common.EFromTo.BlobTrash().String() || raw.fromTo == common.EFromTo.FileTrash().String() {
+		cooked.listOfFilesLocation = raw.listOfFilesToCopy
+	} else if raw.fromTo == common.EFromTo.BlobFSTrash().String() {
+		if len(raw.listOfFilesToCopy) > 0 && (len(raw.include) > 0 || len(raw.exclude) > 0) {
+			return cooked, fmt.Errorf("include/exclude flags are not supported for this destination")
```

Comment:
Are they "not supported for this destination" in all cases, or just "not supported for this destinatino when list-of-files is supplied?" The code implies the lattre, but the error message says the former

Reply:
Good catch, they are never supported!

This makes me think though, when the ADLS Gen2 interop happens eventually, maybe we have to keep the current blob implementation so that users can have this kind of functionality, and we can shift to the new directory delete API if no include/exclude is supplied. I'll add a comment.